File: openapi-client/php-dt/src/App/DTO/DenyRuleRecord.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\PhpAttribute as DTA;

/**
 * A complete deny rule record as returned by &#x60;GET /mail/rules&#x60;.  Combines the rule definition fields (&#x60;DenyRuleNew&#x60;) with server-assigned metadata (&#x60;id&#x60; and &#x60;created&#x60;).  The &#x60;id&#x60; value is required by &#x60;DELETE /mail/rules/{ruleId}&#x60;.
 */
class DenyRuleRecord
{
    /**
     * The type of deny rule.
     */
    #[DTA\Data(field: "type")]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $type = null;

    /**
     * The value to match against, interpreted according to &#x60;type&#x60;: a full email address for &#x60;email&#x60;/&#x60;destination&#x60;, a domain name for &#x60;domain&#x60;, or an alphanumeric prefix string for &#x60;startswith&#x60;.
     */
    #[DTA\Data(field: "data")]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $data = null;

    /**
     * The numeric ID of the deny rule, as a string.  Pass this as &#x60;ruleId&#x60; to &#x60;DELETE /mail/rules/{ruleId}&#x60; to remove the rule.
     */
    #[DTA\Data(field: "id")]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $id = null;

    /**
     * The timestamp when the rule was created.
     */
    #[DTA\Data(field: "created")]
    #[DTA\Strategy("DateTime")]
    #[DTA\Validator("DateTime")]
    public \DateTimeInterface|null $created = null;

    /**
     * Optional SMTP username of the mail order to associate this rule with (e.g. &#x60;mb20682&#x60;).  If omitted the first active order is used.  Valid usernames are the &#x60;username&#x60; values returned by &#x60;GET /mail&#x60;.
     */
    #[DTA\Data(field: "user", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $user = null;

}

File: openapi-client/scala-gatling/src/gatling/scala/org/openapitools/client/model/DenyRuleRecord.scala

package org.openapitools.client.model

import java.time.OffsetDateTime

case class DenyRuleRecord (
    /* The type of deny rule. */
    _type: String,
    /* The value to match against, interpreted according to `type`: a full email address for `email`/`destination`, a domain name for `domain`, or an alphanumeric prefix string for `startswith`. */
    _data: String,
    /* The numeric ID of the deny rule, as a string.  Pass this as `ruleId` to `DELETE /mail/rules/{ruleId}` to remove the rule. */
    _id: String,
    /* The timestamp when the rule was created. */
    _created: OffsetDateTime,
    /* Optional SMTP username of the mail order to associate this rule with (e.g. `mb20682`).  If omitted the first active order is used.  Valid usernames are the `username` values returned by `GET /mail`. */
    _user: Option[String]
)
object DenyRuleRecord {
    def toStringBody(var_type: Object, var_data: Object, var_id: Object, var_created: Object, var_user: Object) =
        s"""
        | {
        | "type":$var_type,"data":$var_data,"id":$var_id,"created":$var_created,"user":$var_user
        | }
        """.stripMargin
}
